### PR TITLE
Fixing static analyzer warnings in JetValidation.cc, JetPlusTrackCorrector.cc and PileupJPTJetIdAlgo.cc, ElectronMVAEstimator.cc

### PR DESCRIPTION
**Pull request by uhussain**
@slava77 and @cvuosalo: you requested to watch this, thanks

### CODE REVIEW

File: RecoEgamma/ElectronIdentification/src/ElectronMVAEstimator.cc [hunk 1]
```diff
@@ -101,7 +101,6 @@ double ElectronMVAEstimator::mva(const reco::GsfElectron& myElectron, int nverti
   bool validKF= false;
 
   reco::TrackRef myTrackRef = myElectron.closestCtfTrackRef();
-  validKF = (myTrackRef.isAvailable());
   validKF = (myTrackRef.isNonnull());
```
Comment (cvuosalo) [inline RecoEgamma/ElectronIdentification/src/ElectronMVAEstimator.cc]: I think we want to `and` the two checks:

```
validKF = (myTrackRef.isNonnull() && myTrackRef.isAvailable()); 
```

File: RecoJets/JetProducers/src/PileupJPTJetIdAlgo.cc [hunk 2]
```diff
@@ -199,13 +199,6 @@ float PileupJPTJetIdAlgo::fillJPTBlock(const reco::JPTJet* jet
       double x2 = (detavar+dphivar-det)/2.;
       
       
-  // Energy fraction in cone
- 
-      ffrac01 = ffrac01/(*jet).energy();
-      ffrac02 = ffrac02/(*jet).energy();
-      ffrac03 = ffrac03/(*jet).energy();
-      ffrac04 = ffrac04/(*jet).energy();
-      ffrac05 = ffrac05/(*jet).energy();
```
Comment (cvuosalo) [inline RecoJets/JetProducers/src/PileupJPTJetIdAlgo.cc:192]: These five variables serve no purpose at all in the code. They are calculated successively but never read. I think all five should be entirely deleted from the file.

File: RecoJets/JetPlusTracks/plugins/JetPlusTrackCorrector.cc [hunk 3]
```diff
@@ -1385,8 +1385,6 @@ if( trBgOutOfVertex.size() == 0 ) return mScale;
 
     EnergyOfBackgroundCharged       = M_PI*mConeSize*mConeSize*EnergyOfBackgroundCharged;
     ResponseOfBackgroundCharged     = M_PI*mConeSize*mConeSize*ResponseOfBackgroundCharged;
-    NumberOfBackgroundChargedVertex = M_PI*mConeSize*mConeSize*NumberOfBackgroundChargedVertex;
-    NumberOfBackgroundChargedCalo   = M_PI*mConeSize*mConeSize*NumberOfBackgroundChargedCalo;
```
Comment (cvuosalo) [inline RecoJets/JetPlusTracks/plugins/JetPlusTrackCorrector.cc:1389]: It looks like these two variables can be deleted entirely from this file. Also, please delete the commented-out lines where they are used.

**Review by uhussain — COMMENTED** [review]
All the dead store/dead assignment errors raised by the static analyzer in these files have been fixed.

**Pull request merged**